src/Datadog.Trace/PlatformHelpers/AzureAppservices.cs

```diff
@@ -120,11 +214,16 @@ private string GetSubscriptionId(IDictionary environmentVariables)
             return null;
         }
 
-        private string GetVariableIfExists(string key, IDictionary environmentVariables)
+        private string GetVariableIfExists(
+            string key,
+            IDictionary environmentVariables,
+            Action<string> optionalExistsAction = null)
         {
             if (environmentVariables.Contains(key))
             {
-                return environmentVariables[key]?.ToString();
+                var value = environmentVariables[key]?.ToString();
```

Comment:
With any dictionary, calling `Contains` and then accessing the indexer are two steps that can be consolidated into one. With `IDictionary` in particular, `Contains` should return null when the key doesn't exist. We have `GetValueOrDefault<T>` and `TryGetValue<T>` extension methods to be safe (in case there's a misbehaving `IDictionary` implementation).

I suggest rewriting this method to something like this:
```csharp
var value = environmentVariables.GetValueOrDefault<string>(key);

if (value != null && optionalExistsAction != null)
{
    optionalExistsAction.Invoke(value);
}

return value;
```

Reply:
Updated.